metrics-operator/controllers/analysis/controller.go

```diff
@@ -136,9 +144,69 @@ func (a *AnalysisReconciler) evaluateObjectives(ctx context.Context, res map[str
 		analysis.Status.Warning = true
 	}
 	analysis.Status.Pass = eval.Pass
+	go a.reportResultsAsPromMetric(eval, analysis)
 	return a.updateStatus(ctx, analysis)
 }
 
+func (a *AnalysisReconciler) reportResultsAsPromMetric(eval evalType.AnalysisResult, analysis *metricsapi.Analysis) {
+	f := analysis.Spec.From.String()
+	t := analysis.Spec.To.String()
+	labelsAnalysis := prometheus.Labels{
+		"name":      analysis.Name,
+		"namespace": analysis.Namespace,
+		"from":      f,
+		"to":        t,
+	}
+	if m, err := a.Metrics.AnalysisResult.GetMetricWith(labelsAnalysis); err == nil {
+		m.Set(eval.GetAchievedPercentage())
+	} else {
+		a.Log.Error(err, "unable to set value for analysis result metric")
+	}
+	// expose also the individual objectives
+	for _, o := range eval.ObjectiveResults {
+		name := o.Objective.AnalysisValueTemplateRef.Name
+		ns := o.Objective.AnalysisValueTemplateRef.Namespace
+		labelsObjective := prometheus.Labels{
+			"name":               name,
+			"namespace":          ns,
+			"analysis_name":      analysis.Name,
+			"analysis_namespace": analysis.Namespace,
+			"key_objective":      fmt.Sprintf("%v", o.Objective.KeyObjective),
+			"weight":             fmt.Sprintf("%v", o.Objective.Weight),
+			"from":               f,
+			"to":                 t,
+		}
+		if m, err := a.Metrics.ObjectiveResult.GetMetricWith(labelsObjective); err == nil {
+			m.Set(o.Value)
+		} else {
+			a.Log.Error(err, "unable to set value for objective result metric")
+		}
+	}
+}
+
+//nolint:ineffassign,staticcheck
```

Comment:
This is for the linting to not complain about the two calls to `err = prometheus.Register(_)`. It screams that we should do something with the error returned.